pandas/tests/series/test_analytics.py

```diff
@@ -1825,6 +1882,27 @@ def test_reshape_2d_return_array(self):
         expected = x.reshape((-1, 1))
         assert_almost_equal(result, expected)
 
+    def test_reshape_bad_kwarg(self):
+        a = Series([1, 2, 3, 4])
+
+        msg = "'foo' is an invalid keyword argument for this function"
+        tm.assertRaisesRegexp(TypeError, msg, a.reshape, (2, 2), foo=2)
+
+        msg = "reshape\(\) got an unexpected keyword argument 'foo'"
+        tm.assertRaisesRegexp(TypeError, msg, a.reshape, a.shape, foo=2)
+
+    def test_numpy_reshape(self):
+        a = Series([1, 2, 3, 4])
+
+        result = np.reshape(a, (2, 2))
+        expected = a.values.reshape(2, 2)
+        tm.assert_numpy_array_equal(result, expected)
+        self.assertTrue(type(result) is type(expected))
+
+        result = np.reshape(a, a.shape)
+        tm.assert_series_equal(result, a)
+        self.assertTrue(type(result) is type(a))
```

Comment:
line 1903. use ``assert_series_equal`` compares virtually everything, no need to ``self.asserTrue(type(result) is type(a))`` which is completly non-idiomatic anyhow, you would simply use ``self.assertIsInstance``; my point is its not necessary